File: 프로그래머스/lv2/42842. 카펫/카펫.java
import java.util.*;

class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int sum = brown + yellow;

        for (int i = 1; i <= sum; i++) {
            int col = i;
            int row = sum / col;

            if (col > row)
                continue;

            if ((row - 2) * (col - 2) == yellow) {
                answer[0] = row;
                answer[1] = col;
                break;
            }
        }

        return answer;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    
    xy = brown + yellow
    for x in range(1 , xy + 1) :
        if (xy / x) % 1 == 0 :
            y = xy / x
            if x >= y :
                if 2*x + 2*y == brown + 4 :
                    answer = [x,y]
                    
    return answer